Проект, выполненный по заданию sprint-1

ORLOVN opened the pull request.

ORLOVN committed:
Проект, выполенный по заданию sprint-1

ORLOVN committed:
Проект, выполенный по заданию sprint-1

khitrind commented on .browserslistrc:
```diff
@@ -0,0 +1,2 @@
+> 5%
+last 2 versions
```
[EOF](https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log) 
https://eslint.org/docs/rules/eol-last

khitrind commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+/node_modules/*
+/dist/*
+!/dist/_redirects
```
EOF

khitrind commented on server.js:
```diff
@@ -0,0 +1,17 @@
+const path = require('path');
+
+const app = express();
+const PORT = 3000;
```
Всегда удобно иметь возможность прокинуть порт снаружи)

`const port = process.env.PORT || PORT;`

khitrind commented on src/core.js:
```diff
@@ -0,0 +1,148 @@
+
+// Инструменты для валидации форм. Bad-функция вернет правдивое значение (строку или true), если поле ввода не прошло
+// валидацию.
+export const validTools ={
```
```suggestion
export const validTools = {
```

khitrind commented on src/core.js:
```diff
@@ -0,0 +1,148 @@
+// валидацию.
+export const validTools ={
+    beautifyPhone: function (str) { //функция для приведения телефона к расивому формату (888) 888-8888
+        str = str.replace(/[^0-9]/g, '');
```
Лучше не модифицировать аргумент функции, а завести новую переменную, так будет меньше шансов запутаться в будущем

khitrind commented on src/core.js:
```diff
@@ -0,0 +1,148 @@
+        return true;
+    }
+}
+export function TemplateCompile(tmpl,tmplData){
```
Функции относящиеся к шаблонизатору, лучше вынести в отдельный модуль, в ином случае файл превратится в мешанину, в которой будет сложно разобраться

khitrind commented on src/index.css:
```diff
@@ -0,0 +1,36 @@
+
+    /* Support for IE. */
+    font-feature-settings: 'liga';
+}
```
EOF

khitrind commented on src/modules/chats/components/chat-pane/chat-pane.tmpl.js:
```diff
@@ -0,0 +1,63 @@
+    </div>
+    <div class="chat-pane__main-pane">
+        <div class="chat-pane__messages-pane">
+            <div class="chat-pane__date-division">
```
Списки принято верстать при помощи тега `<ul>`

https://htmlacademy.ru/blog/boost/frontend/semantics

khitrind commented on src/modules/chats/components/chat-pane/chat-pane.tmpl.js:
```diff
@@ -0,0 +1,63 @@
+                    The CSSStyleSheet interface represents a single CSS stylesheet, and lets you inspect and modify the list of rules contained in the stylesheet. It inherits properties and methods from its parent, StyleSheet.
+                    </div>
+                    <div class="chat-pane__time">
+                    19:32 <span class="material-icons">done_all</span>
```
Для времени подойдет тег `<time>`

khitrind commented on src/modules/chats/components/chatlist/dolldata.js:
```diff
@@ -0,0 +1,31 @@
+export const dollData = {
+    chatlist:[
+    {"id":"id0000001",
+    "name":"Паша",
+    "avatar":"",
+    "unread":5,
+    "time":"10:40",
+    "lastmessege": "Ха, прикольно"
+  },
+  {id: "id0000002",
+    "name":"Илья",
+    "avatar":"",
+    "unread":0,
+    "time":"10:40",
+    "lastmessege": "Вы: Не могу найти информацию"
+  },
+  { id:"id0000003",
+    "name":"Рабочая группа",
+    "avatar":"",
+    "unread":12,
+    "time":"Ср",
+    "lastmessege": "Изображение"
+  },
+  {id:"id0000004",
+    "name":"Квартиросъемщик",
+    "avatar":"",
+    "unread":0,
+    "time":"Пн",
+    "lastmessege": "Шумят очень сильно"
+  }
+]}
```
От форматирования, достаточно сильно зависит, наглядность вашего кода

```suggestion
export const dollData = {
    chatlist: [
        {
            'id': 'id0000001',
            'name': 'Паша',
            'avatar': '',
            'unread': 5,
            'time': '10:40',
            'lastmessege': 'Ха, прикольно',
        },
        {
            id: 'id0000002',
            'name': 'Илья',
            'avatar': '',
            'unread': 0,
            'time': '10:40',
            'lastmessege': 'Вы: Не могу найти информацию',
        },
        {
            id: 'id0000003',
            'name': 'Рабочая группа',
            'avatar': '',
            'unread': 12,
            'time': 'Ср',
            'lastmessege': 'Изображение',
        },
        {
            id: 'id0000004',
            'name': 'Квартиросъемщик',
            'avatar': '',
            'unread': 0,
            'time': 'Пн',
            'lastmessege': 'Шумят очень сильно',
        },
    ],
};

```

khitrind commented on src/modules/signup/signup.css:
```diff
@@ -0,0 +1,155 @@
+.signup__pane-name {
+    margin-left: 25px;
+    margin-top: 17px;
+    font-family: Roboto;
```
Гораздо удобней задать font-family и базовые размеры для всей страницы, а не конкретным элементам

khitrind commented on src/modules/signup/signup.css:
```diff
@@ -0,0 +1,155 @@
+}
+.signup__signin-link a {
+    text-decoration-line: none;
+    color: #EAEAEA;
```
Цвета удобно вынести в css константы. Тогда при рефакторинге придется вносить изменение всего в одном месте

khitrind commented on src/modules/signup/signup.css:
```diff
@@ -0,0 +1,155 @@
+
+    height: 2.5rem;
+    margin-bottom: 0;
+    border: none !important;
```
Лучше избегать использования `!important` в коде, скорее всего того же результат можно добиться при помощи специфичности селекторов 

https://developer.mozilla.org/ru/docs/Web/CSS/Specificity

khitrind commented on src/modules/signup/signup.js:
```diff
@@ -0,0 +1,38 @@
+    handleFormSubmit(event) {
+        event.preventDefault() // Просим форму не отправлять данные самостоятельно
+        let test = true;
+        test &= validTools.validate(this.loginLabel,validTools.badLogin(this.login.value));
```
Лучше избегать дублирования в коде. В данном случае сделать это можно довольно просто:

```

const validators= {
    email: () => {},
    lpgin: () => {},
    ...
};

return inputs.every(({name, value}) => validators[name](value));
```

khitrind reviewed (changes requested):
Привет, спасибо за вашу работу, все выглядит очень не плохо, но нкоторые вещи я бы предложил немного исправить:

- хорошо бы уйти от использования important в стилях;
- разделить js код на отдельные файлы (валидаторы, шаблонизатор);
- избавиться от дублирования (валидация инпутов);

ORLOVN committed:
Отформатирован README, приведен к красивому виду.

ORLOVN committed:
Отредактрован server.js согласно комментариям: добавлена возможность подхватить порт из внешней среды.
Также добавлен newline for EOF

ORLOVN committed:
Во все файлы, где не было, был добавлен newline for EOF.
Отформатировал dolldata.js, согласно комментариям. Переименовал dolldata.js в dummydata.js и переменную dummyData соответственно.

ORLOVN committed:
* Применил Prettier форматирование к core.js (добавлены пробелы где надо, везде использованы двойные кавычки, декларация функции с большим кол-вом аргументов разбито на несколько строк и проч.).
* Исправил функцию beautifyPhone - убрал модифицирование аргумента.
* Удалил лишние функции в core.js.
* Функции относящиеся к шаблонизации выведены в отдельный файл utils/tmplfuncs.js, согласно комментариям.
* Функции относящиеся к валидации форм и утилиты для форм (beautifyPhone) вынесены в отдельный файл utils/validtools.
* Устранил дублирование кода при валидации формы в модуле signin и signup. Сделать прям как в комментарии не получилось, потому что в некоторых случаях передача только значения поля не достаточна, валидация некоторых полей связаны с другими полями (repassword и password например) и валидация полей типа текст (имя, фамилия и проч) должна передавать результат с именем поля (Например: Поле Фамилия не должно быть пустым) - так необходимо в виду того, что label связанный с input используется для выведения результата валидании и, следовательно имя поля должно быть сохранено

ORLOVN committed:
* подправил все css: убрал лишние назначения Roboto. Ввел css константы.
* в chatlist.tmpl.js и в chat-pane.tmpl.js заменил некоторые div на таги time и ul/li по семантическому признаку.

ORLOVN committed:
* добавил css константы для error.css (для страниц 500 и 400)

ORLOVN committed:
* подправил ошибки

ORLOVN committed:
* подправил ошибки

khitrind reviewed (approved):
Спасибо за внесенные правки и удачи в следующем спринте!)

ORLOVN committed:
Merge branch 'main' into sprint-1
